src/pages/en/guides/cms/contentful.md

```diff
@@ -0,0 +1,538 @@
+      date: new Date(item.fields.date).toLocaleDateString(),
+    },
+  }));
+  return pages;
+}
+---
+```
+
+`slug`, the property inside `params`, matches the name of the dynamic route ( `[slug].astro`).
+
+The `props` object passes three properties to the page:
+- title (a string)
+- content (a rich text Document converted to HTML using `documentToHtmlString`)
+- date (formatted using the `Date` constructor)
+
+Finally, you can use the page `props` to display your blog post.
+
+```astro title="src/pages/posts/[slug].astro" ins={21,23-32}
+---
+import { contentfulClient } from "../../lib/contentful";
+import { documentToHtmlString } from "@contentful/rich-text-html-renderer";
+import type { blogPost } from "../../lib/contentful";
+
+export async function getStaticPaths() {
+  const { items } = await contentfulClient.getEntries<blogPost>({
+    content_type: "blogPost",
+  });
+  const pages = items.map((item) => ({
+    params: { slug: item.fields.slug },
+    props: {
+      title: item.fields.title,
+      content: documentToHtmlString(item.fields.content),
+      date: new Date(item.fields.date).toLocaleDateString(),
+    },
+  }));
+  return pages;
+}
+
+const { content, title, date } = Astro.props;
+---
+<html lang="en">
+  <head>
+    <title>{title}</title>
+  </head>
+  <body>
+    <h1>{title}</h1>
+    <time>{date}</time>
+    <article set:html={content} />
+  </body>
+</html>
+```
+
+Navigate to http://localhost:3000/posts/astro-is-amazing/ to make sure your dynamic route is working!
+
+#### Server side rendering
+
+If you've [opted in to SSR mode](/en/guides/server-side-rendering/#enabling-ssr-in-your-project), you will use a dynamic route that uses a `slug` parameter to fetch the data from Contentful.
+
+Create a `[slug].astro` page in `src/pages/posts`. Use [`Astro.params`](/en/core-concepts/routing/#the-astroparams-object) to get the slug from the URL, then pass that to `getEntries`: 
+
+```astro title="src/pages/posts/[slug].astro"
+---
+import { contentfulClient } from "../../lib/contentful";
+import type { BlogPost } from "../../lib/contentful";
+
+const { slug } = Astro.params;
+
+const data = await contentfulClient.getEntries<BlogPost>({
+  content_type: "blogPost",
+  "fields.slug": slug,
+});
+---
+```
+
+If the entry is not found, you can redirect the user to the 404 page using [`Astro.redirect`](/en/guides/server-side-rendering/#astroredirect).
+
+```astro title="src/pages/posts/[slug].astro" ins={7, 12-13}
+---
+import { contentfulClient } from "../../lib/contentful";
+import type { BlogPost } from "../../lib/contentful";
+
+const { slug } = Astro.params;
+
+try {
+  const data = await contentfulClient.getEntries<BlogPost>({
+    content_type: "blogPost",
+    "fields.slug": slug,
+  });
+} catch (error) {
+  return Astro.redirect("/404");
+}
+---
+```
+
+Next, use `documentToHtmlString` to convert the `content` property from a Document to HTML and the Date constructor to format the date:
```

Comment:
Post is new here, but you haven't talked about it or highlighted it, only mentioned these two conversions. I think you need to say where `post` came from and what it's doing here.

Then, below "referencing post" makes more sense.

```suggestion
Next, use `documentToHtmlString` to convert the `content` property from a Document to HTML, and use the Date constructor to format the date:
```